File: modules/engine/src/Math/Random.cpp
#include <ctime>
#include <randar/Math/Random.hpp>

// Seeds the random number generator with the current time.
void randar::seedRandomWithTime()
{
    std::srand(static_cast<unsigned int>(std::time(0)));
}

// Generates a random float within [0, 1].
float randar::randomFloat()
{
    return std::rand();
}

// Generates a random float.
float randar::randomFloat(float min, float max)
{
    return min + (std::rand() / (RAND_MAX / (max - min)));
}

File: test/Math/Random.js
describe.only('Random', function() {
    it('generates random floats between [0, 1]', function() {
        for (let i = 0; i < 1000; i++) {
            let result = randar.randomFloat();
            assert(result >= 0 && result <= 1);
        }
    });

    it('generates random floats within bounds', function() {
        for (let i = 0; i < 1000; i++) {
            let result = randar.randomFloat(500, 6000);
            assert(result >= 500 && result <= 6000);

            result = randar.randomFloat(0.2, 0.98);
            assert(result >= 0.2 && result <= 0.98);
        }
    });
});
